File: move-zeroes/move-zeroes.go
func moveZeroes(nums []int)  {
    idx := 0;
    for i:= 0 ; i < len(nums) ; i++ {
        if nums[i] != 0 { 
            if idx != i{
                nums[i],nums[idx] = nums[idx],nums[i]
            }
            idx = idx + 1
        } 
    }
}

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public static void swap(int[] arr, int i , int j){
        int temp = arr[j];
        arr[j] = arr[i];
        arr[i] = temp;
    }
    public void moveZeroes(int[] nums) {
       int zero_idx = 0;
       for(int i = 0 ; i < nums.length ; i++){
         if(nums[i] != 0){
            swap(nums,i,zero_idx);
            zero_idx++;
         }
       } 
    }
}